Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+    volume (⋃ i ∈ I, s i) = ∑' (i : ↑I), volume (s ↑i) := by
+  have : ∀ i ∈ I, ∃ t ⊆ s i, MeasurableSet t ∧ volume t = volume (s i) ∧ volume (s i \ t) = 0 := by
+    intro i i_I
+    exact nullmeasurable_measurable_null (hm i i_I)
+  choose! t t_s t_m t_v t_z using this
+  have h_st : ⋃ i ∈ I, s i = (⋃ i ∈ I, t i) ∪ (⋃ i ∈ I, (s i \ t i)) := by
+    -- This is probably not the most elegant proof of this rather trivial fact.
+    refine le_antisymm ?_ ?_
+    · intro x
+      simp only [mem_union, mem_iUnion₂]
+      rintro ⟨i, i_I, x_s⟩
+      rcases em (x ∈ t i) with x_t | x_nt
+      · left ; use i, i_I
+      · right ; use i, i_I
+        rw [mem_diff]
+        constructor <;> assumption
+    · refine union_subset ?_ ?_
+      · exact biUnion_mono (subset_refl I) t_s
+      · refine biUnion_mono (subset_refl I) ?_
+        intro i ?_
+        exact diff_subset
+  have hm_ti : ∀ i ∈ I, MeasurableSet (t i) := by
+    intro i i_I
+    exact t_m i i_I
+  have hm_tu : MeasurableSet (⋃ i ∈ I, t i) :=
+    MeasurableSet.biUnion hc hm_ti
+  have h_null : volume (⋃ i ∈ I, (s i \ t i)) = 0 := by
+    exact (biUnion_null hc).mpr t_z
+  have hv_s : volume (⋃ i ∈ I, s i) = volume (⋃ i ∈ I, t i) := by
+    rw [h_st, union_volume_null hm_tu h_null]
+  have hd_t : I.PairwiseDisjoint t := by
+    refine PairwiseDisjoint.mono_on hd t_s
+  have hv_t : volume (⋃ i ∈ I, t i) = ∑' (i : ↑I), volume (t ↑i) := by
+    exact measure_biUnion (μ := volume) hc hd_t hm_ti
+  rw [hv_s, hv_t]
+  refine tsum_congr ?_
+  rintro ⟨i, i_I⟩
+  rw [t_v i i_I]
+
+/-- We now start constructing the Vitali set.
+
+In the setoid vS, two reals in the interval [0,1] are equivalent
+iff their difference is rational. -/
+instance vS : Setoid { x : ℝ // x ∈ Icc 0 1 } where
+  r := fun x y ↦ (↑ x : ℝ) - (↑ y) ∈ range ((↑) : ℚ → ℝ)
+  iseqv := {
+    refl := by
+      intro x
+      simp only [sub_self, mem_range, Rat.cast_eq_zero, exists_eq]
+    symm := by
+      intro x y
+      simp only [mem_range]
+      rintro ⟨t, h⟩
+      use (-t)
+      simp [h]
+    trans := by
+      intro x y z
+      simp only [mem_range]
+      rintro ⟨t1, h1⟩ ⟨t2, h2⟩
+      use (t1 + t2)
+      simp [h1, h2]
+  }
+
+/-- Make a quotient type vT from the setoid vS. -/
+def vT : Type := Quotient vS
+
+lemma vS_vT_surj : ∀ t : vT, ∃ x : { x : ℝ // x ∈ Icc 0 1 }, ⟦x⟧ = t := by
+  intro t
+  have ⟨x, eq⟩ := Quotient.mk_surjective t
+  use x, eq
+
+/-- Use Classical.choose to make a function vRep from vT to vS. -/
+def vRep : vT → { x : ℝ // x ∈ Icc 0 1 } :=
+  fun t ↦ Classical.choose (vS_vT_surj t)
+
+lemma vRep_spec : ∀ t : vT, ⟦vRep t⟧ = t :=
+  fun t ↦ Classical.choose_spec (vS_vT_surj t)
+
+/-- The Vitali set is the image of vRep. -/
+def vitaliSet : Set ℝ := { x : ℝ | ∃ t : vT, ↑(vRep t) = x }
+
+/-- We now shift the Vitali set using rational numbers in the interval [-1,1]. -/
+def vI : Set ℝ := Icc (-1) 1 ∩ range ((↑) : ℚ → ℝ)
```

Comment:
It seems a bit more natural to write
```suggestion
def vI : Set ℝ := ((↑) : ℚ → ℝ) '' Icc (-1 : ℚ) 1
```
It should also have a more descriptive name. But I have a suspicion this isn't even needed.